frontend/server/handlers/artifacts.ts

```diff
@@ -239,6 +254,111 @@ function getGCSArtifactHandler(options: { key: string; bucket: string }, peek: n
   };
 }
 
+function getVolumeArtifactsHandler(options: { bucket: string; key: string }, peek: number = 0) {
+  const { key, bucket } = options;
+  return async (req: Request, res: Response) => {
+    let filePath;
+    let volumeMount;
+    try {
+      // get ml-pipeline-ui pod namespace
+      const SERVER_NAMESPACE = serverInfo.getServerNamespace();
+      // get ml-pipeline-ui pod name
+      const POD_NAME = serverInfo.getServerPodName();
+
+      // only check when POD_NAME and serverNamespace can be obtained
+      if (POD_NAME && SERVER_NAMESPACE) {
+        const [pod, err] = await k8sHelper.getPod(POD_NAME, SERVER_NAMESPACE);
+        if (err) {
+          const { message, additionalInfo } = err;
+          console.error(message, additionalInfo);
+          res.status(500).send(message);
+          return;
+        }
+
+        if (!pod) {
+          const message = `Could not get pod ${POD_NAME} in namespace ${SERVER_NAMESPACE}`;
+          res.status(500).send(message);
+          return;
+        }
+
+        // volumes not specified or volume named ${bucket} not specified
+        if (!Array.isArray(pod?.spec?.volumes) || !pod.spec.volumes.find(v => v?.name === bucket)) {
+          const message = `Failed to open volume://${bucket}/${key}, volume ${bucket} not exist`;
+          res.status(400).send(message);
+          return;
+        }
+
+        // find volumes mount
+        if (
+          Array.isArray(pod?.spec?.containers) &&
+          Array.isArray(pod.spec.containers[0]?.volumeMounts)
```

Comment:
nit: it will be more resilient if we find container by name, because istio sidecar injection might create more containers here.

Reply:
when I first design code, I use `ml-pipeline-ui` to match container name. but if users change deployment container name, it will not work.

discussion: 
1. use `ml-pipeline-ui` container name to match, and suppose that deployment container name never be modified.
2. add config named `SERVER_MAIN_CONTAINER_NAME` at [config.js](https://github.com/kubeflow/pipelines/blob/master/frontend/server/configs.ts#L50) which default to `ml-pipeline-ui`. if deployment container name was changed, it is users responsibility to change `SERVER_MAIN_CONTAINER_NAME` env.
3. use containers[0] as server main container. AFAIK, sidecar injected as append container after exist container.

I prefer solution 1 or solution 3, WDYT?